src/Routes.tsx

```diff
@@ -1,26 +1,35 @@
-import NextLink from 'next/link'
-import NextRouter from 'next/router'
+import NextLink, { LinkProps } from 'next/link'
+import NextRouter, { SingletonRouter } from 'next/router'
 import * as React from 'react'
 import { parse } from 'url'
 
 import Route from './Route'
 
+interface NextRouteOptions {
+  shallow: boolean
+}
+
+type FnType = (
+  route: string,
+  params?: any,
+  locale?: string | NextRouteOptions,
```

Comment:
Maybe it would be better to move `locale` to `NextRouteOptions `?
Or is it to big change?

otherwise maybe rename it to `localeOrOptions`  and resolve it in function to `locale` or options based on type?

Reply:
I agree with your thoughts. 
But I tried do not to rewrite the library a lot, just to add types to current functionality. 
I would leave it as is for now and will adjust in my next PRs.